src/Ros_mpGetRobotCalibrationData.c

```diff
@@ -0,0 +1,160 @@
+// Ros_mpGetRobotCalibrationData.c
+
+// SPDX-FileCopyrightText: 2023, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2023, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+#define FILENAME_RBCALIB_DAT    "RBCALIB.DAT"
+#define PATH_TO_RBCALIB_DAT     "MPRAM1:0\\RBCALIB.DAT"
+
+
+LONG Ros_mpGetRobotCalibrationData(ULONG file_no, MP_RB_CALIB_DATA *rData)
+{
+#if defined (YRC1000) || defined (YRC1000u)
+    // on these controllers we can forward to mpGetRobotCalibrationData(..)
+    return mpGetRobotCalibrationData(file_no, rData);
+
+#elif defined (DX200) || defined (FS100)
+    //The download and parsing of the DAT file is expensive. I only want to
+    //perform it once. So let's cache the data in an array.
+    //
+    //This is an array of POINTERS. We'll only allocate a slot if that calibration
+    //file is configured.
+    static MP_RB_CALIB_DATA* calibrationFiles[MAX_ROBOT_CALIBRATION_FILES];
+    
+    if (file_no == 0) //first time in, let's parse the whole file
+    {
+        int ret;
+        int fd;
+        BOOL bRet;
+
+        const int SIZEOFBUFFER = 256;
+        char buffer[SIZEOFBUFFER];
+
+        rcutils_string_array_t splitSpace = rcutils_get_zero_initialized_string_array();
+        rcutils_string_array_t splitComma = rcutils_get_zero_initialized_string_array();
+
+        bzero(calibrationFiles, sizeof(calibrationFiles));
+
+        //make sure to remove any old copy that may be out there
+        mpRemove(PATH_TO_RBCALIB_DAT); //dont care if this fails
+
+        //save from CMOS to DRAM
+        ret = mpSaveFile(MP_DRV_ID_DRAM, "", FILENAME_RBCALIB_DAT);
+        if (ret != OK)
+        {
+#error raise alarm
+            return ERROR;
+        }
+
+        fd = mpOpen(PATH_TO_RBCALIB_DAT, O_RDONLY, 0);
+        if (fd < 0)
+        {
+#error raise alarm
+            return ERROR;
+        }
+
+        while (TRUE) //unknown file size. this will `break` when done with file
+        {
+            //---------------------------------------------------------------------
+            bRet = FileUtilityFunctions_ReadLine(fd, buffer, SIZEOFBUFFER); //  //RBCALIB x
+
+            if (bRet && strstr(buffer, "//RBCALIB"))
+            {
+                rcutils_split(buffer, ' ', g_motoros2_Allocator, &splitSpace);
+
+                int fileNo = atoi(splitSpace.data[1]);
```

Comment:
Pretty sure this should be `fileNo - 1`

Reply:
[a97720f](https://github.com/Yaskawa-Global/motoros2/pull/49/commits/a97720f937a20b0ccd076ddfd000c71b039be331)